src/transport/mqtt/mod.rs

```diff
@@ -225,18 +225,35 @@ impl<S> MqttClient<S> {
         Ok(())
     }
 
-    async fn mark_as_sent(&self, id: &RetentionId) -> Result<(), Error>
+    async fn mark_sent(
+        &self,
+        id: RetentionId,
+        reliability: Reliability,
+        notice: Token<AckOfPub>,
+    ) -> Result<(), crate::Error>
     where
         S: StoreCapabilities,
     {
-        match id {
-            RetentionId::Volatile(id) => {
-                self.volatile.mark_sent(id, true).await;
+        match reliability {
+            // Since it's Unreliable we will never know the broker received it
+            Reliability::Unreliable => {
+                self.mark_received(&id).await?;
             }
-            RetentionId::Stored(id) => {
-                if let Some(retention) = self.store.get_retention() {
-                    retention.mark_as_sent(id).await?;
+            Reliability::Guaranteed | Reliability::Unique => {
+                match id {
+                    RetentionId::Volatile(id) => {
+                        self.volatile.mark_sent(&id, true).await;
+                    }
+                    RetentionId::Stored(id) => {
+                        if let Some(retention) = self.store.get_retention() {
+                            retention.mark_as_sent(&id).await?;
```

Comment:
Why didn't we call all methods like `mark_sent`? have they a different behavior?
```suggestion
                            retention.mark_sent(&id).await?;
```